### ISSUE TITLE
Error loading rsparse after install

### ISSUE THREAD

**Issue by mscbuck** [dselivanov/rsparse#25]
Hi all, 

I've been receiving this error when trying to load rsparse

1. Have tried installing from CRAN and it appears to install fine, but when loading with library(rsparse)....

`> library(rsparse)
Error: package or namespace load failed for ‘rsparse’ in dyn.load(file, DLLpath = DLLpath, ...):
 unable to load shared object '/Library/Frameworks/R.framework/Versions/3.6/Resources/library/rsparse/libs/rsparse.so':
  dlopen(/Library/Frameworks/R.framework/Versions/3.6/Resources/library/rsparse/libs/rsparse.so, 6): Library not loaded: @rpath/Volumes/SSD-Data/Builds/R-dev-web/QA/Simon/packages/el-capitan-x86_64/Rlib/3.6/float/libs/float.so
  Referenced from: /Library/Frameworks/R.framework/Versions/3.6/Resources/library/rsparse/libs/rsparse.so
  Reason: image not found`

I have confirmed that the float package has been installed (both doing so through CRAN, and have tried through devtools). 

When trying to install rsparse through github, I get the following error:

`* installing *source* package ‘rsparse’ ...
** using staged installation
checking whether the C++ compiler works... yes
checking for C++ compiler default output file name... a.out
checking for suffix of executables... 
checking whether we are cross compiling... configure: error: in '/private/var/folders/wg/zm1l13hs4_j80f2yndrd1z_w0000gp/T/Rtmpu6XVG0/R.INSTALLac747f212ad4/rsparse':
configure: error: cannot run C++ compiled programs.
If you meant to cross compile, use '--host'.
See 'config.log' for more details
ERROR: configuration failed for package ‘rsparse’
* removing ‘/Library/Frameworks/R.framework/Versions/3.6/Resources/library/rsparse’
Error in i.p(...) : 
  (converted from warning) installation of package ‘/var/folders/wg/zm1l13hs4_j80f2yndrd1z_w0000gp/T//RtmpVN3D91/fileac29105bb95d/rsparse_0.3.3.2.tar.gz’ had non-zero exit status`

Usually with above errors this has come down to having to reinstall command line tools, but I have done so and I still get the above

______________________________
sessionInfo()
R version 3.6.0 (2019-04-26)
Platform: x86_64-apple-darwin15.6.0 (64-bit)
Running under: macOS Mojave 10.14.5

Matrix products: default
BLAS:   /System/Library/Frameworks/Accelerate.framework/Versions/A/Frameworks/vecLib.framework/Versions/A/libBLAS.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/3.6/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

loaded via a namespace (and not attached):
 [1] Rcpp_1.0.1        rstudioapi_0.10   magrittr_1.5      usethis_1.5.0     devtools_2.0.2    pkgload_1.0.2     R6_2.4.0          rlang_0.3.4.9003 
 [9] tools_3.6.0       pkgbuild_1.0.3    sessioninfo_1.1.1 cli_1.1.0         withr_2.1.2       remotes_2.0.4     assertthat_0.2.1  digest_0.6.19    
[17] rprojroot_1.3-2   crayon_1.3.4      processx_3.3.1    callr_3.2.0       fs_1.3.1          ps_1.3.0          curl_3.3          testthat_2.1.1   
[25] memoise_1.1.0     glue_1.3.1        compiler_3.6.0    desc_1.2.0        backports_1.1.4   prettyunits_1.0.2

**Comment by dselivanov** [comment]
Hi. Thanks for report. I'm traveling at the moment, will be able to take a
look next week.

ср, 10 июл. 2019 г., 0:48 Michael Clutterbuck <notifications@github.com>:

**Comment by dselivanov** [comment]
The issue with cran version is that you install 'pre-build' package. It seems there is a bug/issue with `float` package that it - see [here](https://www.r-project.org/nosvn/R.check/r-release-osx-x86_64/float-00check.html) and [here](https://www.r-project.org/nosvn/R.check/r-release-osx-x86_64/float-00install.html). Related discussion is [here](https://github.com/wrathematics/float/pull/21#issuecomment-366113428) @snoweye @wrathematics any ideas on how to properly fix this in `rsparse`/`float`?

@mscbuck as a solution I would recommend to install `rsparse` and `float` from source - `install.packages(c('float', 'rsparse'), type = 'source')`. **But** as you've noticed you have an issue with compilers toolkit. 

1. Please install headers as suggested [here](https://stackoverflow.com/a/56364797/1069256):
```
open /Library/Developer/CommandLineTools/Packages/macOS_SDK_headers_for_macOS_10.14.pkg
```
2. After that install **clang-7** (for R>=3.6) and gfortran-6.1 from [official CRAN tools from Mac OS](https://cran.r-project.org/bin/macosx/tools/)  and update/create your ~/.R/Makevars:

```
CXX11 = /usr/local/clang7/bin/clang++
CXX =  /usr/local/clang7/bin/clang++
CC =  /usr/local/clang7/bin/clang
FC = /usr/local/gfortran/bin/gfortran
```

3. 
```
`install.packages(c('float', 'rsparse'), type = 'source')`
```

**Comment by mscbuck** [comment]
Thank you much! Can confirm that I was able to get it working with those changes. Thank you for the quick reply!

**Comment by snoweye** [comment]
See if [this float](https://github.com/snoweye/float/commit/4be7a95251baa4d2903f88810a43be81c6bca773) and [this rsparse](https://github.com/snoweye/rsparse/commit/4f457134d9ff4d447107f81af6272c7e2bfc1e9b) fix the mac osx problems or not. Thanks.

**Comment by dselivanov** [comment]
should be fixed with rsparse `0.3.3.4` - we link to static float lib (i've tried on my my and it works).
